Segmentation fault when supplying credentials for proxy

### System Information
OS: Linux kali 6.0.0-kali6-amd64 https://github.com/saucelabs/forwarder/pull/1 SMP PREEMPT_DYNAMIC Debian 6.0.12-1kali1 (2022-12-19) x86_64 GNU/Linux
go version: go1.19.4 linux/amd64
Commit: 661276909bf3c5f2ba643c769af051c529702776

### Details
Apperently there is a bug which will lead to a segmentation fault when supplying credentials for a proxy and the PACResolver returns DIRECT. When trying to connect to an address which should be handled by a proxy according to the PAC the supplied credentials are correctly applied. When starting the forwarder without credentials parameter the DIRECT connection succeeds.

```
forwarder proxy -p ./test.js --address :8888 --verbose --credentials username:password@upstream.lan:8080
2023/01/20 13:00:24 [proxy] INFO: Using PAC proxy
2023/01/20 13:00:24 [proxy] INFO: Localhost proxying mode: deny
2023/01/20 13:00:24 [proxy] INFO: PROXY server listen address=[::]:8888 protocol=http
2023/01/20 13:00:28 [proxy] DEBUG: accepted connection from 127.0.0.1:49392
2023/01/20 13:00:28 [proxy] DEBUG: waiting for request: 127.0.0.1:49392
2023/01/20 13:00:28 [proxy] DEBUG: attempting to establish CONNECT tunnel: www.google.de:443
2023/01/20 13:00:28 [pac] DEBUG: FindProxyForURL("http://www.google.de:443", "") -> "DIRECT"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xb762d9]

goroutine 39 [running]:
github.com/saucelabs/forwarder.(*CredentialsMatcher).MatchURL(0xc000399980, 0x0)
        /home/kali/forwarder/credentials.go:100 +0x39
github.com/saucelabs/forwarder.(*HTTPProxy).pacProxy(0xc0002c5080, 0x1?)
        /home/kali/forwarder/http_proxy.go:220 +0xd8
github.com/google/martian/v3.(*Proxy).connect(0xc00037f680, 0xc000357300)
        /home/kali/go/pkg/mod/github.com/saucelabs/martian/v3@v3.0.0-20221227135800-27a8c8fc6193/proxy.go:716 +0x37
github.com/google/martian/v3.(*Proxy).handleConnectRequest(0xc00037f680, 0xc0003ac7c0?, 0xc000357300, 0xc0002b8448?, 0xc0003a6470, {0xe4bf78?, 0xc0002b8448?})
        /home/kali/go/pkg/mod/github.com/saucelabs/martian/v3@v3.0.0-20221227135800-27a8c8fc6193/proxy.go:437 +0x1268
github.com/google/martian/v3.(*Proxy).handle(0xc00037f680, 0xc0003ac740, {0xe4bf78, 0xc0002b8448}, 0xc0003a6470)
        /home/kali/go/pkg/mod/github.com/saucelabs/martian/v3@v3.0.0-20221227135800-27a8c8fc6193/proxy.go:561 +0x75f
github.com/google/martian/v3.(*Proxy).handleLoop(0xc00037f680, {0xe4bf78, 0xc0002b8448})
        /home/kali/go/pkg/mod/github.com/saucelabs/martian/v3@v3.0.0-20221227135800-27a8c8fc6193/proxy.go:290 +0x617
created by github.com/google/martian/v3.(*Proxy).Serve
        /home/kali/go/pkg/mod/github.com/saucelabs/martian/v3@v3.0.0-20221227135800-27a8c8fc6193/proxy.go:261 +0x30d
```

PAC File

```
function FindProxyForURL(url, host) {
  if ( dnsDomainIs (host, ".google.de")) {
	return "DIRECT";
  }
  return "PROXY upstream.lan:8080";
}
```

Request
```bash
export http_proxy=http://127.0.0.1:8888
export https_proxy=http://127.0.0.1:8888
export no_proxy=localhost,127.0.0.0/8,::
curl https://www.google.de/
curl: (56) Proxy CONNECT aborted
```

--- a/credentials.go
+++ b/credentials.go
@@ -88,7 +88,7 @@ func NewCredentialsMatcher(credentials []*HostPortUser, log Logger) (*Credential
 
 // MatchURL adds standard http and https ports if they are missing in URL and calls Match function.
 func (m *CredentialsMatcher) MatchURL(u *url.URL) *url.Userinfo {
-	if m == nil {
+	if m == nil || u == nil {
 		return nil
 	}
 

